caffe2/core/tensor.h

```diff
@@ -70,7 +70,7 @@ class CAFFE2_API Tensor final {
   explicit Tensor(at::Device device)
       : impl_(c10::make_intrusive<TensorImpl, UndefinedTensorImpl>(
             Storage::create_legacy(device),
-            c10::computeDispatchKey(at::device(device).layout(at::kStrided)),
+            c10::computeDispatchKey(typeMetaToScalarType(get_default_dtype()), at::kStrided, device),
```

Comment:
The newly added use of `get_default_dtype()` here looks very questionable. What's going on here?

Reply:
Also this is caffe2::Tensor, I am supremely doubtful we should be respecting default dtype on this constructor!

Reply:
Ah - well, it's definitely easier to fix that now :P

Reply:
changed it to pass in `c10::nullopt` and leave it up to `computeDispatchKey` to figure out the default. Don't want this PR to change any behaviors, not fixing the dtype handling for now.